nova/cmd/manage.py

```diff
@@ -652,6 +652,138 @@ def online_data_migrations(self, max_count=None):
+            return 2
+
+        max_faults = int(max_faults)
+        if max_faults < 0:
+            print(_("Must supply a positive value for max_faults"))
+            return 2
+        if max_faults > db_const.MAX_INT:
+            print(_('max faults must be <= %(max_value)d') %
+                  {'max_value': db_const.MAX_INT})
+            return 2
+
+        ctxt = context.get_admin_context()
+        try:
+            cell_mappings = objects.CellMappingList.get_all(ctxt)
+        except db_exc.CantStartEngineError:
+            print(_('Failed to connect to API DB so aborting this deletion '
+                    'attempt. Please check your config file to make sure that '
+                    '[api_database]/connection is set and run this '
+                    'command again.'))
+            return 3
+
+        table_to_rows_deleted = {}
+        if until_complete and verbose:
+            sys.stdout.write(_('Deleting') + '..')  # noqa
+
+        interrupt = False
+
+        if all_cells:
+            # Sort first by cell name, then by table:
+            # +--------------------------------+-------------------------+
+            # | Table                          | Number of Rows Deleted  |
+            # +--------------------------------+-------------------------+
+            # | cell0.instance_faults          | 1                       |
+            # | cell1.instance_faults          | 5                       |
+            # | cell2.instance_faults          | 2                       |
+            # ...
+            def sort_func(item):
+                cell_name, table = item[0].split('.')
+                return cell_name, table
+            print_sort_func = sort_func
+        else:
+            cell_mappings = [None]
+            print_sort_func = None
+        for cell_mapping in cell_mappings:
+            # If all_cells=False, cell_mapping is None
+            with context.target_cell(ctxt, cell_mapping) as cctxt:
+                cell_name = cell_mapping.name if cell_mapping else None
+                try:
+                    while True:
+                        rows_deleted, table_name = \
+                            db.delete_excessive_instance_faults(
+                                cctxt, max_rows=max_rows,
+                                max_faults=max_faults)
+                        if rows_deleted:
+                            if cell_name:
+                                table_name = cell_name + '.' + table_name
+                            table_to_rows_deleted.setdefault(table_name, 0)
+                            table_to_rows_deleted[table_name] += rows_deleted
+                        # Stop deleting if there are no more rows to delete or
+                        # we do not run until all rows are deleted
+                        if rows_deleted < max_rows or not until_complete:
+                            break
+                        if verbose:
+                            sys.stdout.write('.')
```

Comment:
This feels weird. Could probably get replaced with `print('.', end='')`.

I see, that's prior art in the file. Weird.